pkg/controller/gitopsconfig/controller.go

```diff
@@ -183,24 +204,40 @@ func (r *Reconciler) createJob(jobtype string, instance *gitopsv1alpha1.GitOpsCo
 	job, err := util.CreateJob(mergedata)
 	if err != nil {
 		log.Error(err, "unable to create job manifest from merge data", "mergedata", mergedata)
-		return err
+		return "", err
 	}
 	err = controllerutil.SetControllerReference(instance, &job, r.scheme)
 	if err != nil {
 		log.Error(err, "unable to the owner for job", "job", job)
-		return err
+		return "", err
 	}
 
 	log.Info("Creating a new Job", "job.Namespace", job.Namespace, "job.Name", job.Name)
 	err = r.client.Create(context.TODO(), &job)
 	if err != nil {
 		log.Error(err, "unable to create the job", "job", job)
-		return err
+		return "", err
 	}
-	return nil
+	time.Sleep(5 * time.Second)
+	var result batchv1.Job
+	err = r.client.Get(context.TODO(),
+		types.NamespacedName{Name: job.Name, Namespace: job.Namespace},
+		&result)
+
+	if err != nil {
+		log.Info("Error in GetJob:", err)
+		return "", err
+	}
+	if result.Status.StartTime != nil {
+		instance.Status.StartTime = result.Status.StartTime.Format(time.RFC3339)
+		if result.Status.CompletionTime != nil {
+			instance.Status.EndTime = result.Status.CompletionTime.Format(time.RFC3339)
+		}
+	}
+	return getJobStatus(result.Status.Succeeded, result.Status.Failed), nil
```

Comment:
This is not a correct way to implement the `state` in GitOpsConfig status field. It is important that  `state` field must be set dynamically, changing quickly in reaction to changes in job status. The code above has the following main issues:

 1. It loads job status into GitOpsConfig status only once, and does not dynamically detect status of the job. For example, if the job takes 90 seconds to finish, the GitOpsConfig status will show `state` "INPROGRESS" even after the job finished.
 2. It blocks the main thread of controller for 5 seconds (l.221) doing nothing. This blocks the controller from working on its main responsibilities, of reconciling GitOpsConfig objects. There must be a better way to dynamically check status of a job than to wait idly.

My suggestion is to find a way to use Kubernetes Go client library to subscribe to job status change events, so that we can be informed about them immediately when they happen, and react by changing the GitOpsConfig status. If that is not possible, we might be able to at least find a way to monitor the job to detect changes in its status.

Reply:
> My suggestion is to find a way to use Kubernetes Go client library to subscribe to job status change events, so that we can be informed about them immediately when they happen, and react by changing the GitOpsConfig status.

This is called a "watch". Kubernetes client-go or the operator-sdk can create a "watch" to stream k8s changes in real time.